File: app/src/main/java/com/leory/badminton/adapter/MainPagerAdapter.java
package com.leory.badminton.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.List;

/**
 * Describe : 首页PagerAdapter
 * Author : zhuly
 * Date : 2019-05-22
 */
public class MainPagerAdapter extends FragmentPagerAdapter {
    private List<Fragment> fragments;

    public MainPagerAdapter(FragmentManager fm, List<Fragment> fragments) {
        super(fm);
        this.fragments = fragments;
    }

    @Override
    public Fragment getItem(int i) {
        return fragments.get(i);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }
}

File: app/src/main/java/com/leory/badminton/adapter/MainPagerAdapter.kt
package com.leory.badminton.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Describe : 首页PagerAdapter
 * Author : zhuly
 * Date : 2019-05-22
 */
class MainPagerAdapter(fm: FragmentManager, private val fragments: List<Fragment>) : FragmentPagerAdapter(fm) {

    override fun getItem(i: Int): Fragment = fragments[i]

    override fun getCount(): Int = fragments.size
}
